raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -647,58 +633,60 @@ contract TokenNetwork is Utils {
         ChannelSettled(channel_identifier);
     }
 
-    /// @dev Returns the unique identifier for the channel participant balance data.
-    /// @param channel_identifier The channel identifier - mapping key used for `channels`.
-    /// @param participant Address of the channel participant whose balance data
-    /// is referenced by the key.
+    /// @dev Returns the unique identifier for the channel
+    /// @param participant Address of a channel participant.
     /// @param partner Address of the channel partner.
-    /// @return Unique identifier for the participant's balance data.
-    function getBalanceDataKey(uint256 channel_identifier, address participant, address partner)
+    /// @return Unique identifier for the channel.
+    function getChannelIdentifier(address participant, address partner)
         pure
-        internal
-        returns (bytes32 data)
+        public
+        returns (bytes32)
     {
-        return keccak256(channel_identifier, participant, partner);
+        // Lexicographic order of the channel addresses
+        // This limits the number of channels that can be opened between two nodes to 1.
+        if (participant < partner) {
```

Comment:
Why not move the check for `participant == partner` in here and revert if they are the same?

Reply:
I could also move all the `require(participant != 0x0);` checks there